oap-server/ai-pipeline/src/main/proto/ai_http_uri_recognition.proto

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.apache.skywalking.oap.server.ai.pipeline.grpc";
+
+service HttpUriRecognitionService {
+    rpc fetchAllPatterns(HttpUriRecognitionSyncRequest) returns (HttpUriRecognitionResponse) {}
+    rpc recognize(stream HttpUriRecognitionRequest) returns (HttpUriRecognitionResponse) {}
+}
```

Comment:
@Superskyyy These two services, 
- `fetchAllPatterns` for syncing existing patterns. This would be called during OAP booting and call periodically every minute. No server-side operation, just asking for the existing patterns.
- `recognize` is for shipping the raw URIs and expecting to get a response soon. 

**To be discussed**, @Superskyyy @wankai123  considering the frequency of fetchAllPatterns is high(every minute). Should we consider two things
1. Remove HttpUriRecognitionResponse of `recognize`, as the next fetch/sync would come soon enough.
2. Would the AI server side provide some kind of pattern candidate dataset ID? The OAP could hold that in the memory,  and only expect the response of updated patterns if the AI server has new identified results. @Superskyyy Is this possible?

Reply:
1 sounds very reasonable, a simple acknowledge will be enough and fully rely on fetchall for the patterns. 

For 2, could you please explain a bit more on what is the "pattern candidate dataset ID" with an example? Is it the PatternID <-> (ActualPattern, ActualPatternRegex) mapping I previously mentioned?

Reply:
Yes this is possible. A version number can be maintained on algorithm side.